mirrord/layer/src/socket/ops.rs

```diff
@@ -1594,26 +1594,56 @@ pub(super) fn getifaddrs() -> HookResult<*mut libc::ifaddrs> {
         Err(io::Error::from_raw_os_error(result))?;
     }
 
-    // Head of the ipv6 interface addresses list.
-    // Should be freed with `libc::freeifaddrs` before returning.
-    let mut ipv6_head: *mut libc::ifaddrs = std::ptr::null_mut();
+    // Count entries for new_list_start malloc
+    let mut entry_count = 0;
+    let mut count_head: *mut libc::ifaddrs = original_head;
+    unsafe {
+        while let Some(ifaddr) = count_head.as_mut() {
+            entry_count += 1;
+            count_head = ifaddr.ifa_next;
+        }
+    }
+
+    // Allocate new list so we can safely free the original list later
+    let new_list_start: *mut libc::ifaddrs = unsafe {
+        libc::malloc((mem::size_of::<libc::ifaddrs>() as libc::size_t) * entry_count)
```

Comment:
In theory, `malloc` should be harmless, but we've been avoiding using it.

@aviramha came up with a clever solution around this "problem" in a similar PR [#938](https://github.com/metalbear-co/mirrord/pull/938). Would require hooking the `freeifaddrs` to check if the pointers are ours, or if they are users'.

Reply:
nope, since `freeifaddrs` calls `free` underneath we should be good (as long as we use malloc)

Reply:
And as long as the user is also using `malloc`.